python/mlcroissant/mlcroissant/_src/structure_graph/base_node.py

```diff
@@ -53,13 +62,27 @@ class Node:
     ctx: Context = dataclasses.field(default_factory=Context)
     id: str = dataclasses.field(default_factory=generate_uuid)
     name: str | None = None
-    parents: list[Node] = dataclasses.field(default_factory=list)
+    parents: list["Node"] = dataclasses.field(default_factory=list)
```

Comment:
Out of curiosity, why is this needed?

Reply:
I removed `from __future__ import annotations` in order for [`dataclasses.field`](https://docs.python.org/3/library/dataclasses.html#dataclasses.Field)'s types to be populated with types and not strings. Without this annotation, we need to use strings.